tools/azsdk-cli/Azure.Sdk.Tools.Cli/Commands/CommandFactory.cs

```diff
@@ -0,0 +1,68 @@
+using System.CommandLine;
+using System.Text.Encodings.Web;
+using System.Text.Json.Serialization;
+using System.Text.Json;
+using Azure.Sdk.Tools.Cli.Contract;
+using System.Reflection;
+using Microsoft.Azure.Pipelines.WebApi;
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+using ModelContextProtocol.Protocol.Types;
+
+namespace Azure.Sdk.Tools.Cli.Commands
+{
+    public class CommandFactory
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly ILogger<CommandFactory> _logger;
+        private readonly ILoggerFactory _loggerFactory;
+
+        private readonly RootCommand _rootCommand;
+
+        public CommandFactory(IServiceProvider serviceProvider, ILoggerFactory loggerFactory)
```

Comment:
instead of taking IServiceProvider so you can use it in create instances of all of the MCPTool implementations, leave that in your IServiceCollection registrations and take `IEnumerable<MCPTool> tools` as a parameter

```suggestion
        public CommandFactory(IEnumerable<MCPTool> tools, ILoggerFactory loggerFactory)
```

Reply:
I get where you're coming from. We may still end up doing that, but I believe I have this thing working the way we want, only registering the tools that are actually registered. Just gotta work through the reflection in `CreateRootCommand` and `HostServerTool.CreateAppBuilder`